index.d.ts

```diff
@@ -79,44 +79,44 @@ declare namespace megajs {
         label: string;
         owner?: string;
         nodeId?: string;
-        loadedFile?: string;
+        attributes: JSON;
         downloadId: string;
+        timestamp?: number;
         directory: boolean;
         favorited: boolean;
-        timestamp?: number;
+        loadedFile?: string;
         key: Nullable<Buffer>;
         name: Nullable<string>;
-        attributes: JSON;
-        constructor(opts: FileOpts);
-        static fromURL(opt: FileOpts | string, extraOpt?: Partial<FileOpts>): File;
+        get createdAt(): number;
         static unpackAttributes(at: Buffer): void | JSON;
+        static fromURL(opt: FileOpts | string, extraOpt?: Partial<FileOpts>): File;
         static defaultHandleRetries(tries: number, error: err, cb: errorCb): void;
-        get createdAt(): number;
-        loadAttributes(cb: BufferString): this;
+        constructor(opts: FileOpts);
+        loadAttributes(cb: BufferString): Promise<string | this>;
```

Comment:
It does not return a string: `loadedNode` is a `File` object. If you call `loadAttributes` on `https://mega.nz/file/handler#key` or `https://mega.nz/folder/handler#key` then it returns `this`, but if you call it on `https://mega.nz/folder/handler1#key/file/handler2` it returns the file from the `handler2`. That's meant to make easy to work with links that target files in shared folders (something that Mega used not to support, but now they support it quite well in their web interface).

Reply:
lol for some reason I thought the promise was resolving with `this.loadedFile` which was of type string, I'll fix it are there any other method with wrong type?

Reply:
That's fine. I checked here and `uploadAttribute` [also accept strings as `type`](https://mega.js.org/docs/api#uploadattributetype-bufferstream-callback). `mkdir` should return `Promise<MutableFile>`.